### PR TITLE
Make CSVs downloadable

### PR DESCRIPTION
**Pull request by sknep**
This pull request addresses the request to make CSVs feel more like a download, begun here https://github.com/CMSgov/CMCS-DSG-DSS-Certification-Staging/pull/83#issuecomment-984893036

Behind the scenes, this PR adds a [new Jekyll layout](https://github.com/CMSgov/CMCS-DSG-DSS-Certification-Staging/blob/make-csvs-downloadable/_layouts/csv.html) that outputs to .csv (rather than the typical .html), and we're using it to generate the downloadable CSVs. The [original csv files](https://github.com/CMSgov/CMCS-DSG-DSS-Certification-Staging/tree/make-csvs-downloadable/_data) in `_data` are just source files, and Jekyll doesn't typically export that data raw. In order to tell Jekyll where to export these files, each Outcomes subfolder has files named "output-csv-cms.csv" ([example](https://github.com/CMSgov/CMCS-DSG-DSS-Certification-Staging/blob/make-csvs-downloadable/Outcomes%20and%20Metrics/Pharmacy%20Benefit%20Management%20(PBM)%20%26%20Point%20of%20Sale%20(POS)/output-csv-cms.csv)) and "output-csv-state.csv" ([example](https://github.com/CMSgov/CMCS-DSG-DSS-Certification-Staging/blob/make-csvs-downloadable/Outcomes%20and%20Metrics/Pharmacy%20Benefit%20Management%20(PBM)%20%26%20Point%20of%20Sale%20(POS)/output-csv-state.csv)), each of which pairs up a source data file and the intended export location. (A few of these Outcomes don't have CMS tables intentionally, and so those have been omitted.) The source data files are still used to generate the in-page tables as before.

**This pull request changes...**
- CSV download links will now automatically trigger a download on click, rather than open in a new window
- CSV download links now point to the file that is current with the branch or repo (no more waiting until a deploy to "main" for them to work)
- Some CSV data files were renamed so that they all have either "CMS-Required" or "State-Specific", so that they won't need to be renamed in the future.
- Assigning the source data files before the page's first H1 triggered a change to the H1 display that made it look smaller. I've now assigned source data files just before they're used in the page, so the H1s stay the same size.

**Steps to manually review and verify this change...**
1. Review the [Outcomes and Metrics folders](https://github.com/CMSgov/CMCS-DSG-DSS-Certification-Staging/tree/make-csvs-downloadable/Outcomes%20and%20Metrics) to see the new .csv files next to the readme.md files for each outcome. These are placeholder files that tell Jekyll what data to use and where to output it. For easier reasoning about later, they are exported to a /downloads folder on build, using the exact same filename as the source file. It should make it simple to download, edit, and upload a new version of the same file. Keeping the filenames consistent wherever they're referred to is going to be essential, but now that it's all set up, I think it will be easy to maintain.

### This pull request can be merged when…
- [x] The above pull request description is complete
- [x] The pull request author has tested the change successfully
- [x] The pull request author has assigned a CMS reviewer
- [x] The change has been reviewed and approved by CMS

### CODE REVIEW

File: _layouts/csv.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{% assign source = page.source | remove: ".csv" | replace: " ", "_" %}
+{%- for row in site.data[source] -%}
+    {%- if forloop.first -%}
+        {%- for pair in row -%}
+          "{{ pair[0] }}"{%- unless forloop.last %},{% endunless -%}
+          {% if forloop.last %}
+          {% endif %}
```
Comment (mgwalker) [inline _layouts/csv.html:7]: What's this do?

File: _layouts/csv.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        {%- endfor -%}
+    {%- endif -%}
+{% raw %}
+{% endraw %}
```
Comment (mgwalker) [inline _layouts/csv.html:11]: Same question here. I don't know Jekyll well enough so to me it just looks like an empty block. 😛

**Review by mgwalker — APPROVED** [review]
A couple of questions below just to satisfy my curiosity. I'd also suggest deleting the `.DS_Store` files that got committed and adding `.DS_Store` to `.gitignore` so it doesn't end up checked in again. But this is very neat!

**Review by nickarrow — APPROVED** [review]
Good to go, removed .ds_store files.

**Comment by nickarrow** [comment]
@sknep hmmmmm, merged the PR into staging, but the download link is still opening in the browser. Is it working for you?

**Comment by sknep** [comment]
Looking... I don't see the latest changes either. May I have admin access to the Staging repo to double check the github pages branch? I'm wondering if the Staging repo is building the "main" branch still.

File: _layouts/csv.html (hunk 2 above)
Comment (sknep) [reply _layouts/csv.html:11]: It is an empty block! but it's one we want. :) This one, like the one before it, is to get an empty newline after each entry in the array, so we get a valid CSV.

File: _layouts/csv.html (hunk 1 above)
Comment (sknep) [reply _layouts/csv.html:7]: This one is to get an empty newline after the headers print out. You'll notice the rest of the liquid tags have `{%- whatever -%}` — the hyphen basically says "ignore whitespace" (https://shopify.github.io/liquid/basics/whitespace/). Here, we need to preserve a blank new line only on the last header for it to output valid CSV.

**Pull request merged**